Use new paraviewweb

mathturtle opened the pull request:
@jourdain I think this is done now.  I may have missed a few style things when I was refactoring.  Can you play with it and see if you can find anything that is broken/looks wrong?

jourdain commented on lib/app.js:
```diff
@@ -4,13 +4,11 @@ import { Router, Route, hashHistory } from 'react-router';
 
-import vtkWebClientFactory from 'tonic-io/lib/VtkWeb/client/pvw/ClientFactory';
-import VtkWebSmartConnect  from 'tonic-io/lib/VtkWeb/SmartConnect';
+import vtkWebClientFactory from 'paraviewweb/src/IO/WebSocket/ParaViewWebClient';
```
I've changed the API of ParaViewWebClient, so you should pull the latest release of ParaViewWeb (1.0.7) and fix your usage of that class.

Here is an example on how to do it right:

import ParaViewWebClient from 'paraviewweb/src/IO/WebSocket/ParaViewWebClient';

var client = ParaViewWebClient.createClient(connection, [
        'FileListing',
        'MouseHandler',
        'ProxyManager',
        'TimeHandler',
        'ViewPort',
    ], customClient);

mathturtle replied:
What is the variable customClient here?

jourdain replied:
http://kitware.github.io/paraviewweb/api/ParaViewWebClient.html#createClient

jourdain commented on lib/panels/Contour/index.js:
```diff
@@ -154,11 +154,11 @@ export default React.createClass({
-                        return <div key={ idx } className='ContourPanel_line'>
-                                    <i className='fa fa-fw fa-trash-o is-clickable'
+                        return <div key={ idx } className={style.line}>
+                                    <i className={style.trash}
```
You could have name the button .deleteButton instead of .trash. 
Same for the plus above. Those name should describe the element goal not the current icon implementation.

jourdain commented on lib/panels/Dataset/index.js:
```diff
@@ -62,8 +62,8 @@ export default React.createClass({
-                  <div className='DatasetPanelContents'>
-                    <i className='fa fa-sun-o' />
+                  <div className={style.contents}>
+                    <i className={style.sun} />
```
Same here, naming is important... 

I don't know what is the goal of the sun here. Can I replace it with the moon?

jourdain commented on lib/widgets/TogglePanel/index.js:
```diff
@@ -83,7 +82,7 @@ export default React.createClass({
         if (this.props.classIcon)
         {
-          button = <i className={ this.props.classIcon + buttonClass }
+          button = <i className={ [ this.props.classIcon, buttonClass ] }
```
You missed the ".join(' ')"

jourdain commented on style/ContourPanel.mcss:
```diff
@@ -0,0 +1,19 @@
+    align-items: center;
+}
+
+.trash {
```
meaningful name...

jourdain commented on style/ContourPanel.mcss:
```diff
@@ -0,0 +1,19 @@
+    cursor: pointer;
+}
+
+.plus {
```
meaningful name...

jourdain commented on style/DatasetPanel.mcss:
```diff
@@ -0,0 +1,11 @@
+  align-items: center;
+}
+
+.sun {
```
meaningful name...

jourdain commented on style/LightViz.mcss:
```diff
@@ -0,0 +1,11 @@
+:global(.is-clickable) {
```
Do you still need any of that?

jourdain commented on style/MultiSlicePanel.mcss:
```diff
@@ -0,0 +1,22 @@
+.axisLine {
+}
+
+.trash {
```
meaningful name...

jourdain commented on style/MultiSlicePanel.mcss:
```diff
@@ -0,0 +1,22 @@
+    cursor: pointer;
+}
+
+.plus {
```
meaningful name...

jourdain commented on style/ViewDataset.mcss:
```diff
@@ -60,22 +60,22 @@
 }
 
-.ViewDataset__Modules > * {
+.modules > * {
```
Was it possible to have a className for those * ?

mathturtle replied:
I'll look through them and do my best.  Some of them I couldn't since Link does not take a className parameter.

jourdain replied:
Sounds good, I just couldn't remember but that make sense...

jourdain commented:
I've added come inline comments...

mathturtle commented:
I've pushed a new version that should address these.  Plus a few extra bugs I found when going through your feedback.  I probably don't need the global styles in LightViz.mcss anymore....  Do you see anything else?

jourdain commented:
I think that should be it. That looks great! What do you think?

jourdain commented:
Maybe pick the latest version of paraviewweb@1.0.10.

mathturtle commented:
Updated to latest and fixed one more bug I found.  I also removed the style file that contained the global styles like is-clickable since it was no longer needed.
